test/e2e/pod_resource_scaling/pod_resource_scaling_test.go

```diff
@@ -0,0 +1,431 @@
+	_, filename, _, _ := runtime.Caller(0)
+	suiteName = common.GetSuiteName(filename)
+
+	// Initializing kubectl client
+	kClient = k8s.KubeCtl{}
+	Ω(kClient.SetClient()).To(gomega.BeNil())
+
+	// Initializing rest client
+	restClient = yunikorn.RClient{}
+	Ω(restClient).NotTo(gomega.BeNil())
+	yunikorn.EnsureYuniKornConfigsPresent()
+	yunikorn.UpdateConfigMapWrapper(oldConfigMap, "")
+})
+
+var _ = BeforeEach(func() {
+	// Skip if K8s version < 1.32
+	k8sVer, err := kClient.GetKubernetesVersion()
+	Ω(err).NotTo(HaveOccurred())
+	var minVersion *version.Info = &version.Info{
+		Major: "1",
+		Minor: "32",
+	}
+	if k8sVer.Major < minVersion.Major || (k8sVer.Major == minVersion.Major && k8sVer.Minor < minVersion.Minor) {
+		ginkgo.Skip("InPlacePodVerticalScaling requires K8s 1.32+")
+	}
+
+	// Create test namespace
+	ns = "test-" + common.RandSeq(10)
+	_, err = kClient.CreateNamespace(ns, nil)
+	Ω(err).NotTo(HaveOccurred())
+})
+
+var _ = ginkgo.AfterEach(func() {
+	By("Killing all pods")
+	err := kClient.DeletePods(ns)
+	Ω(err).NotTo(HaveOccurred())
+	err = kClient.DeleteNamespace(ns)
+	Ω(err).NotTo(HaveOccurred())
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	yunikorn.RestoreConfigMapWrapper(oldConfigMap)
+})
+
+func verifyYunikornResourceUsage(pod *v1.Pod, resourceName string, value int64) {
+	err = utils.WaitForCondition(func() bool {
+		currentPod, err := kClient.GetPod(pod.Name, ns)
+		if err != nil {
+			return false
+		}
+
+		// Verify scheduler name is yunikorn
+		if currentPod.Spec.SchedulerName != "yunikorn" {
+			return false
+		}
+
+		appID := "yunikorn-" + ns + "-autogen"
+		app, err := restClient.GetAppInfo("default", "root."+ns, appID)
+		if err != nil || app == nil {
+			return false
+		}
+
+		if app.Allocations == nil {
+			return false
+		}
+
+		for _, alloc := range app.Allocations {
+			resVal, exists := alloc.ResourcePerAlloc[resourceName]
+			if !exists {
+				return false
+			}
+
+			if resVal == value {
+				return true
+			}
+		}
+
+		return false
+	}, 30*time.Second, 120*time.Second)
+	Ω(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod should be scheduled by YuniKorn with correct resource(%s) allocation", resourceName))
+}
+
+var _ = ginkgo.Describe("InPlacePodVerticalScaling", func() {
+	ginkgo.It("Pod resources(cpu/memory) resize up", func() {
+		// Create pod with initial resources
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podName,
+				Labels: map[string]string{
+					"app": podName,
+				},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "nginx",
+						Image: "nginx:latest",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								"cpu":    resource.MustParse("100m"),
+								"memory": resource.MustParse("100Mi"),
+							},
+							Limits: v1.ResourceList{
+								"cpu":    resource.MustParse("100m"),
+								"memory": resource.MustParse("100Mi"),
+							},
+						},
+					},
+				},
+			},
+		}
```

Comment:
You don't have to create a pod object directly, usually we just create sleep pods like:
```
  sleepPodConfigs := k8s.SleepPodConfig{NS: ns, Time: 60, CPU: reqCPU, Mem: reqMem}
  sleepObj, podErr := k8s.InitSleepPod(sleepPodConfigs)
  sleepRespPod, err = kClient.CreatePod(sleepObj, ns)
```

Look at other tests.